packages/stripe-integration/src/stripe-upe/stripe-upe-payment-strategy.ts

```diff
@@ -235,27 +231,25 @@ export default class StripeUPEPaymentStrategy implements PaymentStrategy {
         }
     }
 
-    private async _executeWithAPM(methodId: string): Promise<InternalCheckoutSelectors> {
-        const state = this._store.getState();
-        const paymentMethod = state.paymentMethods.getPaymentMethodOrThrow(methodId);
+    private async _executeWithAPM(methodId: string): Promise<PaymentIntegrationSelectors> {
+        const state = this.paymentIntegrationService.getState();
+        const paymentMethod = state.getPaymentMethodOrThrow(methodId);
         const paymentPayload = this._getPaymentPayload(methodId, paymentMethod.clientToken || '');
 
         try {
-            return await this._store.dispatch(
-                this._paymentActionCreator.submitPayment(paymentPayload),
-            );
+            return await this.paymentIntegrationService.submitPayment(paymentPayload);
         } catch (error) {
-            return await this._processAdditionalAction(error, methodId);
+            return this._processAdditionalAction(error, methodId);
         }
     }
 
     private async _executeWithoutRedirect(
         methodId: string,
         shouldSaveInstrument: boolean,
         shouldSetAsDefaultInstrument: boolean,
-    ): Promise<InternalCheckoutSelectors> {
-        const state = this._store.getState();
-        const paymentMethod = state.paymentMethods.getPaymentMethodOrThrow(methodId);
+    ): Promise<PaymentIntegrationSelectors> {
```

Comment:
should returned value be as `PaymentIntegrationSelectors` maybe better to change to void? because we don't use this returned value after call
```suggestion
    ): Promise<void> {
```